source/common/http/header_map_impl.cc

```diff
@@ -190,6 +190,36 @@ void HeaderString::setReference(const std::string& ref_value) {
   string_length_ = ref_value.size();
 }
 
+HeaderString& HeaderString::operator=(HeaderString&& move_value) {
```

Comment:
This is basically the same code as the move constructor. Can we put code in common utility function and just call `freeDynamic()` followed by that function?